### DISCUSSION TITLE
Examples for basic use in go?

### DISCUSSION THREAD

**Discussion by kkukshtel** [cue-lang/cue#2784] [General]
In lieu of proper C/C++ support I'm looking to write a small Go program that just wraps the basic functionality of Cue into some functions that I can export and compile into a DLL using `--buildmode=c-shared`. However I can't really find any good code examples or where I should look for that sort of thing. Any guidance would be appreciated!

**Comment by verdverm** [comment]
https://cuetorials.com/go-api

**Reply by kkukshtel** [reply]
This is helpful but I'm less interested in configuring Cue to some non-default behavior from Go — instead I'd just like some API point that basically replicates default CLI behavior. But I looked at the doc page on what I assume this would be but saw that it isn't filled out yet:

https://cuetorials.com/go-api/replicating/
